### ISSUE TITLE
[Feature Request] Make fmt and spdlog Optional with a -D_USE_EXTERNAL_SPDLOG_FMT Flag

### ISSUE THREAD

**Issue by ava6969** [rapidsai/cudf#16951]
```
if (USE_EXTERNAL_SPDLOG_FMT)
  find_package(fmt CONFIG REQUIRED)
  find_package(spdlog CONFIG REQUIRED)
else()
  include(cmake/thirdparty/get_fmt.cmake)
  include(cmake/thirdparty/get_spdlog.cmake)
endif()
```

This feature proposes adding the ability to optionally use externally provided versions of fmt and spdlog by introducing a CMake flag: -D_USE_EXTERNAL_SPDLOG_FMT. When this flag is set, the build system will skip downloading or configuring fmt and spdlog via rapids-cmake and instead rely on external installations, such as those installed via package managers like vcpkg or system-level package management.
Key Benefits:
    Flexibility: Allows developers to integrate pre-existing installations of fmt and spdlog, avoiding redundant or conflicting installations.
    Compatibility: This makes it easier for projects that already manage dependencies using systems like vcpkg or Conan to integrate with RMM without modifying dependency configurations.
    Customization: Developers can control and customize the versions of fmt and spdlog used in the build process.

Proposed Changes:
    Introduce a new CMake option: -D_USE_EXTERNAL_SPDLOG_FMT=ON.
    When the flag is set to ON, CMake will use find_package(spdlog) and find_package(fmt) to locate the external installations instead of using rapids-cmake to fetch and configure them.
    If the flag is not set, the current behavior of using rapids-cmake to fetch spdlog and fmt will be maintained, ensuring backward compatibility.

Use Case Example:
Developers who have installed spdlog and fmt via vcpkg or other package managers can configure the project as follows:

```
cmake .. -D_USE_EXTERNAL_SPDLOG_FMT=ON \
  -DCMAKE_PREFIX_PATH=/path/to/external/installations \
  -DCMAKE_TOOLCHAIN_FILE=/path/to/vcpkg/scripts/buildsystems/vcpkg.cmake
```
This setup will automatically link the external versions of spdlog and fmt during the build process.

**Comment by bdice** [comment]
Thanks for the issue! There is a related discussion about removing spdlog from RAPIDS public APIs here: https://github.com/rapidsai/build-planning/issues/104

**Comment by davidwendt** [comment]
Is this perhaps closed by https://github.com/rapidsai/cudf/pull/17307 ?

**Comment by vyasr** [comment]
No, not quite yet. There's more follow-up work to be done before we can resolve this issue, but #17307 is part of the solution.

**Comment by vyasr** [comment]
Starting in the 25.04, logging in RAPIDS will be handled by the [new rapids-logger library](https://github.com/rapidsai/rapids-logger). This library will mediate all interactions with spdlog and fmt. That library will also have the ability to use preexisting spdlog and fmt libraries by setting the following CMake options:
```
set(BUILD_SHARED_LIBS ON)
set(RAPIDS_LOGGER_HIDE_ALL_SPDLOG_SYMBOLS OFF)
set(RAPIDS_LOGGER_FMT_OPTION "EXTERNAL_FMT")
```

The last piece will be https://github.com/rapidsai/rapids-cmake/pull/790, which will allow rapids-cmake to find those libraries instead of being overridden by the patches.
